dftimewolf/lib/collectors/grr_hunt.py

```diff
@@ -54,8 +63,52 @@ def _CreateHunt(
     """
     runner_args = self.grr_api.types.CreateHuntRunnerArgs()
     runner_args.description = self.reason
+
+    if self.match_mode:
+      if self.match_mode.lower() == 'any':
+        match_mode = runner_args.client_rule_set.MATCH_ANY
+      elif self.match_mode.lower() == 'all':
+        match_mode = runner_args.client_rule_set.MATCH_ALL
+      else:
+        self.ModuleError(f'Unknown match mode {self.match_mode}', critical=True)
+
+      runner_args.client_rule_set.match_mode = match_mode
+
+    if self.client_labels:
+      for client_label in self.client_labels.split(','):
+        label_rule = runner_args.client_rule_set.rules.add()
+
+        label_rule.rule_type = label_rule.LABEL
+        label_rule.label.label_names.append(client_label)
+
+    if self.client_operating_systems:
+      client_operating_systems = set(
+          os for os in self.client_operating_systems.lower().split(',')
+          if os in ('win', 'osx', 'linux'))
+
+      if len(client_operating_systems) != len(self.client_operating_systems):
+        self.ModuleError('Invalid client OS values.', critical=True)
```

Comment:
Can we move this check to `SetUp()` ? This makes sure that a valid OS version is passed before anything else executes.

Reply:
Done